### PR TITLE
Add 2023 array methods to target features

### LINKED ISSUE #52220: Provide `lib`/`target` suggestions for `findLast`/`findLastIndex`
Once https://github.com/microsoft/TypeScript/pull/49636 goes in, we should ensure that a user gets suggestions for using `findLast`/`findLastIndex` when their output `target` or `lib` is too low.

```ts
function findLastZeroIndex(arr: number[]) {
  return arr.findLastIndex(0);
}
```

We already do this for other missing methods.

### PR DESCRIPTION
**Pull request by OliverRadini**
Fixes #52220

Adds es2023 array methods to target features.

I haven't added any unit tests for this - I couldn't find anything that tests this already, and I'm afraid I don't know enough about the repo to determine where these would make sense. Any pointers would be much appreciated :)

### CODE REVIEW

**Comment by Tobeyyyyy** [comment]
@OliverRadini - Take a look at [indexAt.ts](https://github.com/microsoft/TypeScript/blob/main/tests/cases/compiler/indexAt.ts). You would need to write a similar file for `findLastIndex`. Afterwards, run `hereby runtests-parallel` and `hereby baseline-accept` to create the baseline files.

**Comment by Tobeyyyyy** [comment]
@OliverRadini - Take a look at [indexAt.ts](https://github.com/microsoft/TypeScript/blob/main/tests/cases/compiler/indexAt.ts). You would need to write a similar file for findLastIndex. Afterwards, run hereby runtests-parallel and hereby baseline-accept to create the baseline files.

**Comment by OliverRadini** [comment]
@Tobeyyyyy brilliant, thanks - I'll take a look and add that in

**Comment by OliverRadini** [comment]
@Tobeyyyyy Just taken a look, slightly confused though as it seems as though it may already be covered by the tests in [findLast.ts](https://github.com/microsoft/TypeScript/blob/main/tests/cases/compiler/findLast.ts). I'm probably getting things confused but I'm not sure what I'd add additionally to that.

**Comment by Tobeyyyyy** [comment]
@OliverRadini - Ah, was not aware that a file like that already exists. The first line of the file defines all the targets for which baselines will be created. Right now, the only target in `findLast.ts` is `esnext`. You could now add `es2022` to this list. Running the tests would generate a baseline like `indexAt(target=es2021).errors.txt` which includes the error message.

**Comment by OliverRadini** [comment]
@Tobeyyyyy ah brilliant, thanks. I amended the line as suggested and indeed did see the errors as expected. Is this the normal behaviour for the tests? I added the line, but now there are failing tests. I suppose I don't quite yet understand the workflow. The errors I get are:

> tests/cases/compiler/findLast.ts(1,44): error TS2550: Property 'findLast' does not exist on type 'number[]'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2023' or later.
tests/cases/compiler/findLast.ts(2,51): error TS2550: Property 'findLast' does not exist on type 'string[]'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2023' or later.
tests/cases/compiler/findLast.ts(3,17): error TS2550: Property 'findLast' does not exist on type 'Int8Array'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2023' or later.

So, what I would expect to see after the change I added on a version which doesn't yet have findLast. Is this the expected behaviour?

**Comment by Jack-Works** [comment]
Run > npx hereby baseline-accept

**Comment by OliverRadini** [comment]
@microsoft-github-policy-service agree

**Comment by OliverRadini** [comment]
@Jack-Works Great, thanks - ran that and all good now. Thanks for the help

**Comment by OliverRadini** [comment]
Hopefully this is in a better state now, once I figured out a bit more about how the baseline stuff works 😂

**Review by DanielRosenwasser — APPROVED** [review]

File: src/compiler/utilities.ts [hunk 1]
```diff
@@ -1257,6 +1257,20 @@ export function getScriptTargetFeatures(): ScriptTargetFeatures {
...
+            Uint8Array: ["findLastIndex", "findLast"],
+            Uint8ClampedArray: ["findLastIndex", "findLast"],
+            Int16Array: ["findLastIndex", "findLast"],
+            Uint16Array: ["findLastIndex", "findLast"],
+            Int32Array: ["findLastIndex", "findLast"],
+            Uint32Array: ["findLastIndex", "findLast"],
+            Float32Array: ["findLastIndex", "findLast"],
+            Float64Array: ["findLastIndex", "findLast"],
+            BigInt64Array: ["findLastIndex", "findLast"],
+            BigUint64Array: ["findLastIndex", "findLast"],
```
[the remark this answers is not in the document]
Comment (OliverRadini) [reply src/compiler/utilities.ts:1296]: Ah ok, it'd be good to get those in as well; what changes would be required to additionally add those? Is everything else ready for them to go in here?

[the remark this answers is not in the document]
Comment (ehoogeveen-medweb) [reply src/compiler/utilities.ts:1296]: FWIW, the [Change Array by Copy](https://github.com/tc39/proposal-change-array-by-copy) proposal is still stage 3, whereas [Array find from last](https://github.com/tc39/proposal-array-find-from-last) is a finished proposal (stage 4). I don't know what the policy is on proposals, though.

[the remark this answers is not in the document]
Comment (OliverRadini) [reply src/compiler/utilities.ts:1296]: I'm guessing this is best left out for now; unless anyone is particularly keen that it also goes in. Can always be added later I'm guessing

[the remark this answers is not in the document]
Comment (DanielRosenwasser) [reply src/compiler/utilities.ts:1296]: I don't think those methods are currently in `lib.d.ts`, so we probably shouldn't suggest them yet.

**Pull request merged**